.github/actions/build-cmake-project/action.yml

```diff
@@ -0,0 +1,117 @@
+name: Build CMake Project
+description: Build a CMake project using the specified build type and configuration.
+
+inputs:
+  project-name:
+    description: Name of the CMake project to build.
+    required: true
+  swift-version:
+    description: Built Swift compiler version.
+    required: true
+  enable-caching:
+    description: Enable sccache.
+    required: false
+    default: false
+  debug-info:
+    description: Enable debug information in the build.
+    required: false
+    default: false
+  build-os:
+    description: Build operating system (e.g., "Windows", "Darwin").
+    required: true
+  build-arch:
+    description: Build architecture (e.g., "x86_64", "arm64").
+    required: true
+  os:
+    description: Operating system to build for (e.g., "Windows", "Android").
+    required: true
+  arch:
```

Comment:
Can we name these `host-os` and `host-arch`?

Reply:
I do not think they should, as they can be either `build`, `host` or `target` depending on the job.